### ISSUE TITLE
Issues with MakeMaker tests

### ISSUE THREAD

**Issue by billr1** [Perl-Toolchain-Gang/ExtUtils-MakeMaker#269]
I am building Perl and associated libraries for a cross-compiled environment (qemu x86_64 target) from an x86_64 build machine for the Open Source Yocto project. This is with Perl 5.22.1 (MakeMaker 7.04_01). As part of the verification process, I am running the test cases on the target and get several failures on the MakeMaker tests. t/basic.t is one such example; the other failing tests fail in the same manner.

The error is:

> cpan/ExtUtils-MakeMaker/t/basic
> 1..0 # SKIP cross-compiling and make not available
> No root path(s) specified
 > at t/basic.t line 83.
> ok 1
> ok 2 - teardown
> \# Tests were run but no plan was declared and done_testing() was not seen.
> FAILED--expected 0 tests, saw 2

basic.t contains (in part):

```perl
use ExtUtils::MakeMaker;
   :
   :
use Config; 
use Test::More;
use ExtUtils::MM;
 
plan !MM->can_run(make()) && $ENV{PERL_CORE} && $Config{'usecrosscompile'}
    ? (skip_all => "cross-compiling and make not available")
    : (tests => 171);
use File::Find;
   :
```
I am finding that the call `MM->can_run(make())` returns undef. make() is defined in ExtUtils::MM_Any.pm, but it appears to never be called. In addition, make() relies on the value of $self->MAKE, which is set in init_MAKE(), also defined in ExtUtils::MM_Any.pm and never called.

Now, I can get this code to work by changing it to look like this:

```perl
my $mm = ExtUtils::MM->new();
plan !$mm->can_run($mm->make()) && ...
```
(init_MAKE() gets called from new(), to set the internal value for MAKE which is used when make() is called.)

or this:
```perl
plan !MM->can_run($Config{make}) && ...
```

So, my questions are:

1. How is this intended to work and why did it presumably work when originally written and committed? One potential answer here is that it was never tested in a cross-compile environment, since when $Config{'usecrosscompile'} is not defined, the value for MM->can_run() doesn't matter, as the plan expression will end up always being true.

2. I notice some other tests use MM->can_run($Config{make}). That would work here, too, and solve the problem, but $Config{make} could be different than the value returned by make(); does that really matter for these tests?

As an aside, the cryptic "No root path(s) specified..." error message is caused by this line:
	`END { rmtree $DUMMYINST; }`
which always gets executed, even if no tests are run and $DUMMYINST is not initialized. I changed this line to:
	`END { rmtree $DUMMYINST if defined $DUMMYINST; }`

**Comment by haarg** [comment]
This question was already [answered](http://www.nntp.perl.org/group/perl.makemaker/2016/05/msg3404.html) on the perl.makemaker mailing list.  The bug was fixed in 4a07a3bd18363986112cf2b39dec3c2985353ffb, but that change has not yet made it into perl core.

The `make` sub is coming from `ExtUtils::Test::Utils` in `t/lib`.

You can work around the problem by wrapping the plan call in a `BEGIN { }` block, and not making any other changes.

**Comment by billr1** [comment]
My apologies. When I sent the email to makemakers@ I didn't realize I was sending to a mailing list (that was not clear in the doc); I thought it was just going to the maintainers. If I knew it was a mailing list submission, I would have checked back there.
